exec/java-exec/src/main/java/org/apache/drill/exec/store/base/BaseGroupScan.java

```diff
@@ -0,0 +1,481 @@
+   * plugins generally don't care which node runs which slice of a scan, we
+   * retain only the endpoint count. Later, in {@link #getSpecificScan(int)},
+   * Drill will ask for subscans with minor fragment IDs from 0 to the number of
+   * endpoints specified here.
+   */
+  @Override
+  public void applyAssignments(List<DrillbitEndpoint> endpoints) {
+    endpointCount = endpoints.size();
+  }
+
+  /**
+   * Return the maximum parallelization width. This number is usually the number
+   * of distinct sub-scans this group scan can produce. The actual number of
+   * minor fragments may be less. (For example, if running in a test, the minor
+   * fragment count may be just 1.) In that case, the
+   * {{@link #getSpecificScan(int)} may need to combine logical sub-scans into a
+   * single physical sub scan. That is, a sub scan operator should hold a list
+   * of actual scans, and the scan operator should be ready to perform multiple
+   * actual scans per sub-scan operator.
+   * <p>
+   * For convenience, returns 1 by default, which is the minimum. Derived
+   * classes should return a different number if they support filter push-down
+   * and parallelization.
+   *
+   * @return the number of sub-scans that this group scan will produce
+   */
+  @JsonIgnore
+  @Override
+  public int getMaxParallelizationWidth() { return 1; }
+
+  @JsonIgnore
+  @Override
+  public ScanStats getScanStats() {
+    if (scanStats == null) {
+      scanStats = computeScanStats();
+    }
+    return scanStats;
+  }
+
+  /**
+   * Compute the statistics for the scan used to plan the query. In general, the
+   * most critical aspect is to give a rough estimate of row count: a small row
+   * count will encourage the planner to broadcast rows to all Drillbits,
+   * something that is undesirable if the row count is actually large.
+   * Otherwise, more accurate estimates are better.
+   */
+  protected abstract ScanStats computeScanStats();
+
+  /**
+   * Create a sub scan for the given minor fragment. Override this
+   * if you define a custom sub scan. Uses the default
+   * {@link BaseSubScan} by default.
+   * <p>
+   * The group can may have a single "slice". If so, then the
+   * {@link #getMaxParallelizationWidth()} should have returned 1,
+   * and this method will be called only once, with
+   * <code>minorFragmentId = 0</code>.
+   * <p>
+   * However, if the group can can partition work into sub-units
+   * (let's call them "slices"), then this method must pack
+   * <i><b>n</b></i> slices into <i><b>m</b></i> minor fragments
+   * where <i><b>n</b></i> >= <i><b>m</b></i>.
+   */
+  @Override
+  public abstract SubScan getSpecificScan(int minorFragmentId);
+
+  @JsonIgnore
+  @Override
+  public String getDigest() { return toString(); }
+
+  /**
+   * Create a copy of the group scan with with candidate projection
+   * columns. Calls
+   * {@link BaseScanFactory#groupWithColumns()}
```

Comment:
Thanks for the explanation, in this case, please also remove brackets, since it causes error in JavaDoc: `{@link BaseScanFactory#groupWithColumns()}` -> `{@link BaseScanFactory#groupWithColumns}` here and in JavaDoc for method below.